src/XMakeCommandLine/CommandLineSwitches.cs

```diff
@@ -438,14 +438,22 @@ bool unquoteParameters
             bool parametersStored = false;
 
             // if this is the first time this switch has been detected
-            if (_parameterizedSwitches[(int)parameterizedSwitch].commandLineArg == null)
+            if (_parameterizedSwitches[(int) parameterizedSwitch].commandLineArg == null)
             {
                 // initialize its parameter storage
-                _parameterizedSwitches[(int)parameterizedSwitch].parameters = new ArrayList();
-            }
+                _parameterizedSwitches[(int) parameterizedSwitch].parameters = new ArrayList();
 
-            // save the switch text
-            _parameterizedSwitches[(int)parameterizedSwitch].commandLineArg = commandLineArg;
+                // save the switch text
+                _parameterizedSwitches[(int) parameterizedSwitch].commandLineArg = commandLineArg;
+            }
+            else
+            {
+                _parameterizedSwitches[(int) parameterizedSwitch].commandLineArg = string.Concat(
+                    _parameterizedSwitches[(int)parameterizedSwitch].commandLineArg,
+                    " ",
+                    commandLineArg
+                    );
```

Comment:
Hmm, this will reorder command line parameters so that each kind is in a group, right? That's unfortunate, but still clearly better than the old way.